dataops/dev_cicd_template.yml

```diff
@@ -2,13 +2,45 @@ variables:
   GITLAB_CICD_IMAGE_REPO: gcr.io/getindata-images-public/cicd-dataops
   GITLAB_CICD_IMAGE_TAG: gcp-0.1.1
   BLOB_CONFIG_PATH: blob_args.json
+  GIT_KEY_PATH: TBS
 
 dev:
   image: $GITLAB_CICD_IMAGE_REPO:$GITLAB_CICD_IMAGE_TAG
+  environment: DEVELOPMENT
   stage: execute-dev
   extends:
-    - .setup
+    - .dev_setup
   script:
-    - dp compile --docker-build
+    - dp compile --docker-build --env dev
     - dp deploy --blob-args $BLOB_CONFIG_PATH --datahub-ingest --docker-push
-    - dp publish --key-path $GIT_KEY_PATH
\ No newline at end of file
+  rules:
+    - if: '$CI_COMMIT_BRANCH != "master"'
+
+release-dev:
+  image: $GITLAB_CICD_IMAGE_REPO:$GITLAB_CICD_IMAGE_TAG
+  environment: DEVELOPMENT
+  stage: execute-release
+  extends:
+    - .dev_setup
+  script:
+    - dp compile --docker-build --env dev --docker-tag $CI_COMMIT_TAG
+    - dp deploy --blob-args $BLOB_CONFIG_PATH --datahub-ingest --docker-push
+  rules:
+    - if: '$CI_COMMIT_TAG'
+    - if: '$CI_COMMIT_TAG =~ /^v\d+\.\d+\.\d+/' #expects 'vX.Y.Z'
+
```

Comment:
Adding here another rule `if: '$CI_COMMIT_BRANCH != "master"'` will stop release for dev in case of pushing tag to master.

Reply:
no, the idea is that each release (from master) should be deployed to all envs. I could even add if: '$CI_COMMIT_BRANCH == "master"' everywhere

Reply:
So this is to synchronise master and dev environments in case of release from master?

If yes, this could confuse user to not having the same code on dev branch and on dev env

Reply:
there is no dev branch, there are feature branches. All of them may compete for dev env unfortunately (standard issue).